src/new-components/modal/modal.tsx

```diff
@@ -248,13 +248,15 @@ export function ModalHeader({
             >
                 <Columns space="large" alignY="center">
                     <Column width="auto">{children}</Column>
-                    <Column width="content" exceptionallySetClassName={styles.buttonContainer}>
-                        {typeof button !== 'boolean' ? (
-                            button
-                        ) : button === true ? (
-                            <ModalCloseButton aria-label="Close modal" autoFocus={false} />
-                        ) : null}
-                    </Column>
+                    {button === false || button === null || button === undefined ? null : (
+                        <Column width="content" exceptionallySetClassName={styles.buttonContainer}>
+                            {typeof button === 'boolean' ? (
```

Comment:
It won't make any functional difference, but since we've already eliminated false (or falsy) values from `button`, would checking for `true` explicitly make more sense? Otherwise it could read like we want to render the button even when it's false.

Reply:
In my opinion, testing for `true` would be redundant and less semantic. Above we do a truthy/falsy check and the second condition does a type check (although it's true that testing `true` would match the type too). I think that's more readable than doing another boolean comparison.